### PR TITLE
FHC: use exception-safe versions of ets update functions

### PR DESCRIPTION
**Pull request by michaelklishin**
Otherwise, if the FHC runs into an exception because of an ETS write failure, most code paths begin failing.

This may help with the condition in #8784.

### CODE REVIEW

File: deps/rabbit_common/src/file_handle_cache.erl [hunk 1]
```diff
@@ -1125,13 +1158,13 @@ handle_call({open, Pid, Requested, EldestUnusedSince}, From,
     case needs_reduce(State #fhc_state { open_count = Count + Requested }) of
         true  -> case ets:lookup(Clients, Pid) of
                      [#cstate { opened = 0 }] ->
-                         true = ets:update_element(
+                       safe_ets_update_element(
```
Comment (lhoguin) [inline deps/rabbit_common/src/file_handle_cache.erl:1161]: Probably not important but based on the other changes I would expect `_ = safe_ets...` here.

**Review by lhoguin — APPROVED** [review]
This works as a temporary measure, approved.

**Comment by michaelklishin** [comment]
This addresses several scenarios to the best of our current understanding without introducing any obvious regressions. Let merge it and allow the user to try out a `3.12.2` alpha.

**Comment by michaelklishin** [comment]
Note that the OCI build failures are not related.

**Pull request merged**